### PR TITLE
Initial Pi Zero support

### PR DESCRIPTION
**Pull request by tannewt**
Both SD card and USB don't work. REPL works over UART.

Progress on #5631

### CODE REVIEW

**Comment by tannewt** [comment]
TODO: I have to make a PR to TinyUSB to use a different instruction for the memory barrier.

**Comment by tannewt** [comment]
tinyusb PR is here: https://github.com/hathach/tinyusb/pull/1271

**Comment by tannewt** [comment]
Ok, this is finally ready for review.

File: main.c [hunk 1]
```diff
@@ -813,8 +813,8 @@ int __attribute__((used)) main(void) {
     // no SPI flash filesystem, and we might erase the existing one.
 
     // Check whether CIRCUITPY is available. Don't check if it already hasn't been found.
```
Comment (dhalbert) [inline main.c:815]: I thought I needed to reset into safe mode to get the message to print, but I see that is not the case. So the comment is now a bit misleading.
```suggestion
    // Check whether CIRCUITPY is available.
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Nice work! Just a comment change. But I did not test.

File: main.c (hunk 1 above)
Comment (tannewt) [reply main.c]: The broadcom port can't reset (but may have filesystem issues) so I had a reason to switch this.

**Review by dhalbert — APPROVED** [review]

**Pull request merged**